torch/_dynamo/convert_frame.py

```diff
@@ -218,6 +218,13 @@ def exception_handler(e, code, frame=None, export=False):
 FRAME_COUNTER = 0
 
 
+def is_recompilation(code, cache_size):
+    # code in input_code is not sufficient. For nn module instance methods, we
+    # have per module instance cache, so we need to check the cache size as
+    # well.
+    return code in input_codes and cache_size >= 1
```

Comment:
You need to keep input_codes faithfully aligned with the rest of your changes. If code is now homed elsewhere, the py-side input_codes tracking must reflect that.